### PR TITLE
Issue/592/refactor map features

### LINKED ISSUE #592: Refactor map features
- Change structure of map folder
- Rewrite some files to ts
- Add geolocation to test
- Add my postition feature

### PR DESCRIPTION
**Pull request by Bedpaw**
Check only after #586 

close #592

### CODE REVIEW

File: client/src/map/config.ts [hunk 1]
```diff
@@ -22,4 +22,26 @@ export const mapConfig = {
...
+      },
+      star: {
+        points: 5,
+        radius: 12,
+        radius2: 4,
+        angle: 0,
+      },
+    },
+  },
+  myPosition: {
```
Comment (dbetka) [inline client/src/map/config.ts:42]: myPosition should has the same options available like point. Thanks that you can define any shape in config.
Example below ↓↓↓
```
points: 20,
radius: 10,
angle: 20,
```

File: client/src/map/factories/pointsFeature.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const pointValues = {
+    fill,
+    stroke,
+    points: 20,
+    radius: 10,
+    angle: 20,
+  };
+  if (appearance.shape === pointCategoryUtils.availableShapes.star) {
+    Object.assign(pointValues, {
+      points: 5,
```
Comment (dbetka) [inline client/src/map/factories/pointsFeature.ts:76]: Why it's here and in `map/config.ts`?

File: client/src/map/factories/pointsFeature.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  return color;
+};
+
+const getFinalPoints = (appearance: PointCategoryAppearance, fill: Fill, stroke: Stroke) => {
+  const pointValues = {
+    fill,
+    stroke,
+    points: 20,
+    radius: 10,
+    angle: 20,
```
Comment (dbetka) [inline client/src/map/factories/pointsFeature.ts:72]: Why it's here and in `map/config.ts`?

**Review by dbetka — CHANGES_REQUESTED** [review]
Some based things to fix.
More complex fixes will come later

File: client/src/map/config.ts (hunk 1 above)
Comment (Bedpaw) [reply client/src/map/config.ts:42]: Will change

File: client/src/map/factories/pointsFeature.ts (hunk 2 above)
Comment (Bedpaw) [reply client/src/map/factories/pointsFeature.ts:76]: Will change

File: client/src/map/factories/pointsFeature.ts (hunk 3 above)
Comment (Bedpaw) [reply client/src/map/factories/pointsFeature.ts:72]: Will change

**Pull request closed without merging**